contracts/RMRK/multiresource/RMRKMultiResource.sol

```diff
@@ -344,7 +344,7 @@ contract RMRKMultiResource is
         _beforeTokenTransfer(owner, address(0), tokenId);
 
         // Clear approvals
-        _approve(address(0), tokenId);
+        _approve(address(0), tokenId); // I don't get why we are setting approval for 0x0 if we are also deleting it (assigning it to the same address)
```

Comment:
On 721 the just delete it like this: `delete _tokenApprovals[tokenId];` . 
We cannot do something that simply though, because our approvals are a nested mapping, not a simple one:
`mapping(uint256 => mapping(address => address)) private _tokenApprovals;`
The reason is on top of the definition:
```
// Mapping from token ID to approver address to approved address
// The approver is necessary so approvals are invalidated for nested children on transfer
// WARNING: If a child NFT returns the original root owner, old permissions would be active again
```

Reply:
I answered thinking of transferring.
For burn we could avoid it we assume tokens are not reminted, but I don't think it hurts to have this here.